yt_dlp/extractor/videocampus_sachsen.py

```diff
@@ -78,18 +109,45 @@ class VideocampusSachsenEmbedIE(InfoExtractor):
     ]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        host, video_id, tmp_id, display_id, embed_id = self._match_valid_url(url).group(
+            'host', 'id', 'tmp_id', 'display_id', 'embed_id')
+        webpage = self._download_webpage(url, video_id or tmp_id, fatal=False) or ''
+
+        if not video_id:
+            video_id = embed_id or self._html_search_regex(
+                rf'src="https?://{host}/media/embed.*(?:\?|&)key=([0-9a-f]+)&?',
+                webpage, 'video_id')
 
-        webpage = self._download_webpage(url, video_id)
-        title = self._html_search_regex(r'<img[^>]*title="([^"<]+)"', webpage, 'title', fatal=False)
-        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
-            f'https://videocampus.sachsen.de/media/hlsMedium/key/{video_id}/format/auto/ext/mp4/learning/0/path/m3u8',
-            video_id, 'mp4', 'm3u8_native', m3u8_id='hls')
+        # Title, description from page meta wouldn't be correct if the video is embedded
+        # so separating normal and embedded extraction
+        if not (display_id or tmp_id):
+            title = self._html_search_regex(r'<img[^>]*title="([^"<]+)"', webpage, 'title', fatal=False)
+            description = None
+        else:
+            title = self._html_search_meta(('og:title', 'twitter:title', 'title'), webpage, default=None)
+            description = self._html_search_meta(('og:description', 'twitter:description', 'description'), webpage, default=None)
+
+        # Many pages don't provide HLSs but some do, so ignoring 404s
+        formats = []
+        subtitles = {}
+        try:
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                f'https://{host}/media/hlsMedium/key/{video_id}/format/auto/ext/mp4/learning/0/path/m3u8',
+                video_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=True)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 404:
+                pass
+            else:
+                raise
```

Comment:
```suggestion
            if isinstance(e.cause, compat_HTTPError) and e.cause.code != 404:
                raise
```
avoids `pass`

Reply:
This is wrong. We want it to raise for non-HTTP errors
```suggestion
            if not isinstance(e.cause, compat_HTTPError) or e.cause.code != 404:
                raise
```